### PR TITLE
Make device names valid object paths parts

### PR DESCRIPTION
**Pull request by kalkin**

### CODE REVIEW

File: qubesdbus/device_manager.py [hunk 1]
```diff
@@ -234,7 +235,7 @@ def Detached(self, vm_obj_path):
 
 
 def device_path(vm, dev_class, ident):
-    _id = ident.replace('.', '_')
+    _id = re.sub(r"[^A-Za-z0-9_/]", "_", ident)
     return os.path.join(SERVICE_PATH, dev_class, str(vm.qid), _id)
```
Comment (marmarek) [inline qubesdbus/device_manager.py:239]: While at it, maybe use VM name here? It is now immutable.

File: qubesdbus/device_manager.py [hunk 2]
```diff
@@ -234,7 +235,7 @@ def Detached(self, vm_obj_path):
 
 
 def device_path(vm, dev_class, ident):
-    _id = ident.replace('.', '_')
+    _id = re.sub(r"[^A-Za-z0-9_/]", "_", ident)
```
Comment (marmarek) [inline qubesdbus/device_manager.py:238]: How original ident is saved for making attach/detach calls?

File: qubesdbus/device_manager.py (hunk 1 above)
Comment (kalkin) [reply qubesdbus/device_manager.py:239]: Awesome will do

File: qubesdbus/device_manager.py (hunk 2 above)
Comment (kalkin) [reply qubesdbus/device_manager.py:238]: As a property on the D-Bus object implemeting Device interface.

Comment (kalkin) [reply qubesdbus/device_manager.py:238]: PS: It's part of the `data` dictionary passed at the instantiation of `Device(BUS_NAME, OBJ_PATH, DATA)`

**Review by marmarek — APPROVED** [review]

**Pull request merged**